src/main/java/edu/harvard/iq/dataverse/Dataset.java

```diff
@@ -335,7 +335,7 @@ public DatasetVersion getLatestVersion() {
 
     public DatasetVersion getLatestVersionForCopy() {
         for (DatasetVersion testDsv : getVersions()) {
-            if (testDsv.isReleased() || testDsv.isArchived()) {
+            if (testDsv.isReleased() || testDsv.isArchived() || testDsv.isDeaccessioned()) {
```

Comment:
This looks like it will fix the npe problem, but this method is used in many places and I'm not sure that all of them should include a deaccessioned version. For example - https://github.com/IQSS/dataverse/blob/4c1e069082b740849e3ad0ade41a926407ec52f5/src/main/java/edu/harvard/iq/dataverse/export/ExportService.java#L152 would export info about a deaccessioned version. We might need a separate getLatestPublicVersion method for such cases unless there's some other way to fix.